app/controllers/availabilities_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class AvailabilitiesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @time_ranges = TimeRange.order(:start_time)
+    @availabilities = load_availabilities
+    @start_time = start_time
+    @end_time = end_time
+  end
+
+  def update_availabilities
+    availability_ids = if params.key?(:availability_ids)
+                         params[:availability_ids] else []
+                       end
+    Availability.where(user_id: current_user.id).each do |availability|
+      set(availability,
+          availability_ids.include?(availability.id.to_s))
+    end
+  end
+
+  private
+
+  def set(availability, status)
+    availability.status = status ? 1 : 0
+    availability.save
+  end
+
+  def load_availabilities
+    availabilities = []
+    7.times { |day| availabilities.push(get_availabilities_day(day)) }
+    availabilities
+  end
+
+  def get_availabilities_day(day)
+    availabilities = []
+
+    @time_ranges.each do |time_range|
+      availabilities.push(get_availability(day, time_range.id))
+    end
+    availabilities
+  end
+
+  def get_availability(day, time_range_id)
+    Availability.where(user_id: current_user.id, day: day,
+                       time_range_id: time_range_id)[0] ||
+      create_new_availability(day, time_range_id)
+  end
+
+  def create_new_availability(day, time_range_id)
+    Availability.create(user_id: current_user.id, day: day,
+                        time_range_id: time_range_id)
+  end
+
+  def start_time
```

Comment:
These two methods should probably be put in the model instead? What do you think, do you think it's better to put this here or in the model?

Reply:
which model?

Reply:
Actually nvm can just put here bah. Was thinking the TimeRange model but I think might be weird.

But do change the [0] and [-1] to .first and .last.